docs/source/keywords.rst

```diff
@@ -306,8 +306,32 @@ Architecture Keywords
       * Optimize for HSW architecture
       * ``OFF``
 
-    * * ``Kokkos_ARCH_INTEL_GEN``
-      * Optimize for Intel GPUs Gen9+
+    * * ``KOKKOS_ARCH_INTEL_GEN``
+      * Optimize for Intel GPUs, JIT
```

Comment:
Do we explain somewhere what JIT means and how it differs from other options?

Reply:
No, as far as I know, this is the only occurrence of something JIT related in `Kokkos`. Do you think it's sufficient to just spell it out here?

Reply:
You will need to write a paragraph that explains the alternatives ahead-of-time VS just-in-time and what these entail.
Adding as a follow-up is fine if you prefer

Reply:
Fair enough. I gave it a shot. 🙂